### PR TITLE
Parallel MOF combilation

### PR DESCRIPTION
**Pull request by raandree**
- Adding support for parallel MOF compilation. The number if jobs can be defined with parameter 'MofCompilationTaskCount'. If this parameter is not used, standard serial build is used
  - Starting more than 2 processes loading the same modules can cause issues. This is why the 'RandomWait' parameter is required in parallel build to let the new processes wait to a random number of seconds.
- Adding support for parallel Meta MOF compilation.
- Throwing if the compilation does not succeed so InvokeBuild get the failure status
- Removing 'FilterNode' and 'FilteredNode' references as it is not used and redundant with 'Filter'.
- Simple code reorg and code style alignments

The MOF build still works successful.

### CODE REVIEW

**Comment by gaelcolas** [comment]
Failed build?
Flying today, won't have time to look.

**Comment by raandree** [comment]
Working now, there was still a case sensitivity issue with the roles in the config files.

**Comment by raandree** [comment]
@gaelcolas, any comments on this PR?

File: .build/DSC/ConfigData.build.ps1 [hunk 1]
```diff
@@ -125,6 +125,7 @@ task Compile_Datum_Rsop {
         mkdir -Path $rsopOutputPathVersion -Force | Out-Null
     }
 
+    Write-Build Green "Generating RSOP output for $($configurationData.AllNodes.Count) nodes"
```
Comment (gaelcolas) [inline .build/DSC/ConfigData.build.ps1]: this is good, I should also add a synopsis to the task at some point,

**Comment by raandree** [comment]
Right...

**Comment by raandree** [comment]
This took quite some time:

```
PSDesiredStateConfiguration\node : Specified argument was out of the range of valid values.
Parameter name: times
At C:\Git\Aldi DSC\Operations\RootConfiguration.ps1:16 char:5
+     node $ConfigurationData.AllNodes.NodeName {
+     ~~~~
    + CategoryInfo          : OperationStopped: (:) [PSDesiredStateConfiguration\node], ArgumentOutOfRangeException
    + FullyQualifiedErrorId : System.ArgumentOutOfRangeException,PSDesiredStateConfiguration\node
```

The error was actually caused in RootConfiguration.ps1::51 if "55 -$okMessage.Length" results in a negative number. Fix is on the way...

**Comment by raandree** [comment]
Ping...

**Review by gaelcolas — APPROVED** [review]

**Comment by gaelcolas** [comment]
All good and tested (finally).
Merged

**Pull request merged**